### DISCUSSION TITLE
Trouble installing

### DISCUSSION THREAD

**Discussion by beaver316** [naver/egjs-flicking-plugins#53] [General]
Hi,

I'm by no means an expert with Javascript since I'm primarily a backend developer, however I'm trying to install the plugins for added functionality.

I have setup the standard flicking library by adding the following:

```
<!-- flicking -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/egjs-flicking/4.11.0/flicking.pkgd.min.js" crossorigin="anonymous"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/egjs-flicking/4.11.0/flicking.css" crossorigin="anonymous" />
```

```
<script >
    const flicking = new Flicking("#carousel", {
        align: "center",
        circular: true,     // when viewport fits all cards, keeps looping
        renderOnlyVisible: true,
        easing: x => 1 - Math.pow(1 - x, 3),
        inputType: ["touch", "mouse"]
    });
</script>
```

I want to install the plugin with just CDN, no npm installs or anything like that.

I try adding this code:

```
<!-- flicking plugins -->
<script src="https://cdn.jsdelivr.net/npm/@egjs/flicking-plugins@4.7.1/dist/plugins.js" crossorigin="anonymous"></script>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@egjs/flicking-plugins@4.7.1/dist/flicking-plugins.css">
```

```
<script >
    // my previous flicking initialization here

    flicking.addPlugins(new Autoplay());
</script>
```

I get the error Uncaught ReferenceError: Autoplay is not defined at (index):229:33.

I'm sure I'm missing something simple.

**Comment by malangfox** [comment]
Hello @beaver316.

You can try `flicking.addPlugins(new Flicking.Plugins.AutoPlay())` to resolve this error.
This is a [demo](https://codepen.io/malangfox/pen/oNyxzvB) that may give you a reference.

Try the above solution and let me know if you encounter any additional errors.

**Reply by beaver316** [reply]
Hi malangfox,

Thanks so much, that worked for AutoPlay and Fade. However for Pagination I received the following error:

> Uncaught Error: [Flicking-Pagination] Couldn't find element with the given selector: .flicking-pagination
>     at __proto.init (Pagination.ts:110:13)
>     at Flicking.ts:1450:36
>     at Array.forEach (<anonymous>)
>     at a.addPlugins (Flicking.ts:1450:15)
>     at (index):224:18

This is my code:

```
flicking.addPlugins(
    new Flicking.Plugins.AutoPlay(),
    new Flicking.Plugins.Pagination({
        type: "bullet"
    }),
    new Flicking.Plugins.Fade());
```

Looking at the documentation, for Pagination specifically I need to import pagination.css. I'm not sure if the fact that I don't import this is causing the error but I suspect it. Where is this located on CDN?

**Reply by malangfox** [reply]
This seems to be an error that can occur when there is no div with a `flicking-pagination` class inside the `flicking-viewport` element.
Can you check this [demo](https://codepen.io/malangfox/pen/BavaVJJ) and try including `flicking-pagination` div like in this example?

**Reply by beaver316** [reply]
Thank you so much, that was what I was missing.
